### PR TITLE
Add [vrf_interfaces] resource module

### PR DESCRIPTION
**Pull request by KB-perByte**
##### SUMMARY

 iosxr vrf_interfaces resource module

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

iosxr_vrf_interfaces
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by roverflow — APPROVED** [review]
+1 Pretty straight forward module

File: tests/integration/targets/iosxr_vrf_interfaces/tests/common/_populate_config.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+- name: Populate with simple overridden
+  cisco.iosxr.iosxr_vrf_interfaces:
```
Comment (Ruchip16) [inline tests/integration/targets/iosxr_vrf_interfaces/tests/common/_populate_config.yaml:3]: @KB-perByte should we consider using the `iosxr_config` module here to populate and remove the config? This approach might help us avoid using the `vrf_interfaces` module for setup/teardown in its own tests, ensuring better test isolation and reliability and as we are following this approach in almost all the new resource modules. What do you think?

File: tests/integration/targets/iosxr_vrf_interfaces/tests/common/_remove_config.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+- name: Populate with simple overridden
+  cisco.iosxr.iosxr_vrf_interfaces:
```
Comment (Ruchip16) [inline tests/integration/targets/iosxr_vrf_interfaces/tests/common/_remove_config.yaml:3]: same here as per the last comment

**Review by Ruchip16 — CHANGES_REQUESTED** [review]

File: tests/integration/targets/iosxr_vrf_interfaces/tests/common/_populate_config.yaml (hunk 1 above)
Comment (KB-perByte) [reply tests/integration/targets/iosxr_vrf_interfaces/tests/common/_populate_config.yaml]: Good point, updated the same.

File: tests/integration/targets/iosxr_vrf_interfaces/tests/common/_remove_config.yaml (hunk 2 above)
Comment (KB-perByte) [reply tests/integration/targets/iosxr_vrf_interfaces/tests/common/_remove_config.yaml:3]: Not the same here, it's not required as the vrf name might be different in each case

**Pull request merged**